enforcer/README.md

```diff
@@ -132,10 +149,15 @@ behalf of the Enforcer.
         // return (bool, error)
         //   true - the policy was applied
         //   false - the policy was not applied
-        //   string - a human readable error returned by the plugin. valid 
+        //   string - a human readable error returned by the plugin. valid
         //     if false is returned.
         Apply(id int, data []byte) (bool, string)
 
+COMMENT: I was confused by the name "Apply".  The goal seems to be upsert.
+
+COMMENT: Every policy is named by an integer?  As written, the caller has to keep track of which integers have already been used and create a new one as required.  Since the policy name isn't for consumption by humans, I'd expect the ID to be generated by the system during creation, and then provided by the client on update.  Of course, that would mean splitting Apply into Create/Update.  I expect there's a Read call too.
```

Comment:
This id is simply there so the enforcer and plugin can know which policy is being referred to.  The policy itself has a description, etc. but that doesn't matter at the level of this interface.  The just need to know how to talk to each other about the policy.

That being said, update = unapply old, apply new. We have to understand the impact of potential service interruptions in that universe.  But thats easier to understand then to figure out what update means. 

A read call may become necessary for dealing with the case where the enforcer loses its mind, while the thing the plugin in managing retains its faculties.  But I'd rather make that the plugins' problem

Reply:
I was articulating why this API is problematic for the caller.  I'd recommend splitting it into create and update, which the user of the API will have an easier time with.